src/rtc/core.py

```diff
@@ -152,6 +153,304 @@ def save_as_cog(filename, scratch_dir = '.', logger = None,
+    """Check for the existence and coverage of provided ancillary inputs
+       (DEM) wrt. to the HLS tile. The function
+       also updates the product's dictionary metadata indicating the coverage
+       of each ancillary input wrt. to the input HLS tile
+
+       Parameters
+       ----------
+       check_ancillary_inputs_coverage: bool
+               Flag that enable/disable checks for all ancillary inputs
+               excluding the shoreline shapefile
+       check_shoreline_shapefile: bool
+               Flag that checks for the shoreline shapefile
+       dem_file: str
+               DEM filename
+       geogrid: isce3.product.GeoGridParameters
+               Product's ISCE3 geogrid object
+       metadata_dict: collections.OrderedDict
+               Metadata dictionary
+       logger : logging.Logger, optional
+              Logger object
+    """
+    if logger is None:
+        logger = logging.getLogger('rtc_s1')
+    logger.info(f"Check ancillary inputs' coverage:")
+
+    # file description (to be printed to the user if an error happens)
+    dem_file_description = 'DEM file'
+
+    if not check_ancillary_inputs_coverage:
+
+        # print messages to the user
+        logger.info(f'    {dem_file_description} coverage:'
+                    ' (not tested)')
+
+        # update DSWx-HLS product metadata
+        metadata_dict['DEM_COVERAGE'] = 'NOT_TESTED'
+
+        return
+
+    rasters_to_check_dict = {'DEM': (dem_file_description, dem_file)}
+
+    tile_min_x_utm = geogrid.start_x
+    tile_max_y_utm = geogrid.start_y
+    tile_max_x_utm = geogrid.start_x + geogrid.spacing_x * geogrid.width
+    tile_min_y_utm = geogrid.start_y + geogrid.spacing_y * geogrid.length
+
+    tile_srs = osr.SpatialReference()
+    tile_srs.ImportFromEPSG(geogrid.epsg)
+
+    for file_type, (file_description, file_name) in \
+            rasters_to_check_dict.items():
+
+        # check if file was provided
+        if not file_name:
+            error_msg = f'ERROR {file_description} not provided'
+            logger.error(error_msg)
+            raise ValueError(error_msg)
+
+        # check if file exists
+        if not os.path.isfile(file_name):
+            error_msg = f'ERROR {file_description} not found: {file_name}'
+            logger.error(error_msg)
+            raise FileNotFoundError(error_msg)
+
+        # test if tile is fully covered by the ancillary input
+        # by checking all ancillary input vertices are located
+        # outside of the tile grid.
+        gdal_ds = gdal.Open(file_name, gdal.GA_ReadOnly)
+
+        file_geotransform = gdal_ds.GetGeoTransform()
+        file_projection = gdal_ds.GetProjection()
+        min_x, dx, _, max_y, _, dy = file_geotransform
+        file_width = gdal_ds.GetRasterBand(1).XSize
+        file_length = gdal_ds.GetRasterBand(1).YSize
+
+        del gdal_ds
+
+        max_x = min_x + file_width * dx
+        min_y = max_y + file_length * dy
```

Comment:
it would be great if the variable name more clearly mention this is from ancillary input data.